test/unit/typedQueryBuilderTests.ts

```diff
@@ -1337,17 +1337,72 @@ describe('TypedKnexQueryBuilder', () => {
 
     it('should handle two levels of where exists', (done) => {
         const typedKnex = new TypedKnex(knex({ client: 'postgresql' }));
-        const query = typedKnex.query(User).whereExists(User, (subQuery => {
+        const query = typedKnex.query(User).whereExists(User, (subQuery) => {
             subQuery.whereColumn('status', '=', 'status').whereColumn('id', '<>', 'id');
 
             subQuery.whereExists(User, (subQuery2 => {
                 subQuery2.whereColumn('status', '=', 'status').whereColumn('id', '<>', 'id');
             }));
-        }));
+        });
 
         const queryString = query.toQuery();
         assert.equal(queryString, `select * from "users" where exists (select * from "users" as "subquery0$users" where "subquery0$users"."weirdDatabaseName" = "users"."weirdDatabaseName" and "subquery0$users"."id" <> "users"."id" and exists (select * from "users" as "subquery0$subquery0$users" where "subquery0$subquery0$users"."weirdDatabaseName" = "subquery0$users"."weirdDatabaseName" and "subquery0$subquery0$users"."id" <> "subquery0$users"."id"))`);
 
         done();
     });
+
+    it('should get correct column for nested queries', (done) => {
+        const typedKnex = new TypedKnex(knex({ client: 'postgresql' }));
+        let nestedStatusColumnName;
+        // let nestedUserCategoryYear;
+        const query = typedKnex.query(User)
+            .whereExists(User, (subQuery) => {
+                nestedStatusColumnName = subQuery.getColumnNameWithQuotes('status');
+                // nestedUserCategoryYear = subQuery.getColumnNameWithQuotes('category', 'year');
+            });
+
+        query.toQuery();
+        assert.equal(nestedStatusColumnName, '"subquery0$users"."weirdDatabaseName"');
+        // assert.equal(nestedUserCategoryYear, '"subquery0$users"."category"."year"');
+
+        done();
+    });
+
+    // it('should be able to refer to main query in where exists', (done) => {
+    //     const typedKnex = new TypedKnex(knex({ client: 'postgresql' }));
+    //     const query = typedKnex.query(User);
+    //     query.whereExists(User, (subQuery1) => {
+    //         subQuery1.whereColumn('status', '=', 'status'); // Compares subQuery1 with its parent (query). Would have same result as .whereColumnFirstParent
+
+    //         subQuery1.whereExists(User, (subQuery2) => {
+    //             subQuery2.whereColumnFirstParent('status', '=', 'status'); // Compares subQuery2 with the first parent (query)
```

Comment:
This whole test is commented out because it requires a new feature. I thought that we could get this to work without aliases by adding new methods instead. 
I figured we could first find out how many parents there are through some checks
```
let parents = 0;
if (this.parentTypedQueryBuilder) {
    parents++;
    if (this.parentTypedQueryBuilder.parentTypedQueryBuilder) {
        parents++;
        // etc
    }
}
```
Or whatever else you can think of. 

Then `whereColumnFirstParent` could refer to this subquery's oldest (first) parent which we have calculated. We can throw an error if the required parent doesn't exist with some useful message perhaps. 
Then `whereColumnSecondParent` would refer to the child of the oldest parent, etc.

The alternative would be some way with aliases, although I don't see how that would work with the current setup. Or we can drop this feature entirely. up to you :)

Reply:
Thought about it some more: how about `subQuery2.whereColumn(subQuery2.getColumn('status'), '=', query.getColumn('status'));`?

(Updated in my branch)

Reply:
Awesomesauce! I've added the test and it works. Good stuff 😁